tests/datasets/test_datasets.py

```diff
@@ -126,3 +126,27 @@ def test_demo_replay_dataset():
     dataset = create_dataset(env, train_tasks)
     assert len(dataset[-1].states) == 2
     assert len(dataset[-1].actions) == 1
+
+
+def test_demo_nonoptimal_replay_dataset():
+    """Test demo+nonoptimal-replay dataset creation with Covers env."""
+    # Note that the planning timeout is intentionally set low enough that we
+    # cover some fails to plan from the replays, but not so low that planning
+    # always fails. Also the number of replays is set high enough that we
+    # consistently cover the failure case.
```

Comment:
which failure case, do you mean ApproachFailure

Reply:
@tomsilver question for you

Reply:
yeah ApproachFailure or ApproachTimeout